core/src/main/scala/cr/pulsar/Reader.scala

```diff
@@ -32,6 +34,7 @@ import scala.util.control.NoStackTrace
 trait Reader[F[_], E] {
   def read: Stream[F, Message[E]]
   def read1: F[Option[Message[E]]]
+  def readUntil(timeout: Int, units: TimeUnit): F[Option[Message[E]]]
```

Comment:
Can't this be a `FiniteDuration` instead? It seems both arguments are related.

Reply:
Indeed